server/server.go

```diff
@@ -360,7 +370,56 @@ func (server HTTPServer) reportEndpoint(writer http.ResponseWriter, request *htt
 		Data: rules,
 	}
 
-	err = responses.SendOK(writer, responses.BuildOkResponseWithData("report", report))
+	err := responses.SendOK(writer, responses.BuildOkResponseWithData("report", report))
+	if err != nil {
+		log.Error().Err(err).Msg(responseDataError)
+	}
+}
+
+func (server HTTPServer) singleRuleEndpoint(writer http.ResponseWriter, request *http.Request) {
+	requestRuleID, err := readRuleID(writer, request)
+	if err != nil {
+		return
+	}
+
+	aggregatorResponse, successful := server.fetchAggregatorReport(writer, request)
+	if !successful {
+		return
+	}
+
+	var rule proxy_types.RuleWithContentResponse
+
+	for _, aggregatorRule := range aggregatorResponse.Report {
```

Comment:
would not it be better to query just the one rule, instead of finding them manually there? If it won't be possible, it might be better to refactor this "find-loop" into new function please?

Reply:
agree

Reply:
But if this PR is urgent for UI, we could postpone this optimization since we're not going to have many rules in a while.

Reply:
Ok that's fair. In this case I reword it: `if it is not possible, please create a task with description, that refactoring of this part is needed`.

Reply:
It's possible, but is there really some difference, between running this cycle in aggregator or in smart proxy?

Reply:
yes:
1) it might have been changed on aggregator side, it's aggregator internal thing how to handle requests
2) amount of data transferred will be much less, probably by 1/10 now, 1/50 later (depends on #rules + #hits)

Reply:
but feel free to just create an issue for ^, and let the code as is :)

Reply:
@tisnik created issues https://github.com/RedHatInsights/insights-results-aggregator/issues/896 https://github.com/RedHatInsights/insights-results-smart-proxy/issues/62

Reply:
nice!